src/pygama/flow/fileDB.py

```diff
@@ -0,0 +1,358 @@
+from __future__ import annotations
+
+import json
+import logging
+import os
+import re
+import string
+
+import h5py
+import numpy as np
+import pandas as pd
+from parse import parse
+
+from pygama.lgdo import Array, LH5Store, VectorOfVectors
+
+log = logging.getLogger(__name__)
+
+
+class FileDB:
```

Comment:
Yes, we initially wanted to do that, but DataFrame is weirdly difficult to inherit from. https://pandas.pydata.org/pandas-docs/stable/development/extending.html#extending-subclassing-pandas 

The `self.fileDB.df.query()` syntax isn't ideal, but Jason recommended we just do that instead of figuring out their syntax just to define some extra fields.